app/controllers/subscriptions_controller.rb

```diff
@@ -28,6 +28,18 @@ def create
     render json: { id: subscription.id }, status: status
   end
 
+  def index
+    subscriber = find_subscriber
+
+    subscriptions = Subscription.active.
+      includes(:subscriber_list).
+      where(subscriber: subscriber).
+      order('subscriber_lists.title').
+      as_json(include: :subscriber_list)
+
+    render json: { subscriber: subscriber.as_json, subscriptions: subscriptions }
+  end
```

Comment:
This endpoint seems a touch unusual for an index. You'd expect that for an index you could traverse through all of the subscriptions in the system rather than a particular subscriber (and maybe you could then filter by address or addresses)

I'd suggest that since this all quite a specific case (and could hit growing pains if someone has a lot of subscriptions) that you might want to have this on it's own action within /subscriptions so index is free for more index like usages. 

Not sure quite what to name it as it's not super restful (I think closest rest approach is that we're actually doing a show on subscriber) but could be: `/subscriptions/subscriber/kevin@example.com`

Reply:
I'm seeing this as an `index` of all of a user's subscriptions. To me it doesn't make sense to traverse all subscriptions in the system without at least a user filter. `show` would then be looking at a particular subscription. For other concepts like subscriber lists or subscribers, it makes sense to have an `index` that lists them all (with appropriate pagination of course).

Reply:
I think it does have users outside of just being a particular user. eg if you want to filter by a list and get all the subscriptions associated with it.

Can go with this but it's a bit of a RPC call wrapped onto a rest endpoint so is at risk of causing WTFs.

Reply:
I've given this some more thought and have come up with `GET /subscribers/test@example.com/subscriptions` so that we scope subscriptions to a subscriber. This will also allow us to have endpoints such as `POST /subscribers/test@example.com/address` to change a subscriber's email address.